### PR TITLE
Engagement/jessica/15073 louisiana validation

### PR DESCRIPTION
**Pull request by JessicaWNava**
This PR addresses changes request by LA after their first test message

Test Steps:
1. Make sure the integration tests pass. 
2. Check the email and make sure that the fields are added that they specified other than PID 11.7 which SR does not send over and OBX.7 which we don't send over for this kind of test.

MSH.05.03 : Receiving Application Universal ID type is blank. | This field is required and should be set to 'ISO'
-- | --
– Data Analyzed: : '' |  
MSH.06.02 : Receiving Facility Universal ID is blank. | MSH.6.2 must be 'LADOH' when sending ELR data to LA
– Data Analyzed: : '' |  
MSH.06.03 : Receiving Facility Universal ID type is blank | should be set to 'ISO'
– Data Analyzed: : '' |  
PID.11.07 : REQUEST - Patient Address Type is blank. It should be populated from HL70190 table. | Are you able to provide this field?
– Data Analyzed: : '' |  
PID.11.09 : REQUEST - Patient County/Parish Code failed check. | Are you able to provide this field? Please refer to table HL7 FIPS6-4
– Data Analyzed: : 'Orleans' |  
OBX.07 : REQUIRED - No Reference Range provided. Feedback: Ex: 'Not Detected' | Qualitative Results Reference Ranges: Negative for tests resulting as Positive
– Data Analyzed: : '' | Not Detected for tests resulting as Detected Normal for tests resulting as Abnormal

## Changes
- Addressed LA's concerns with the first test message. 
- Added an integration test for them.

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml [hunk 1]
```diff
@@ -31,13 +31,103 @@ elements:
...
     resource: 'Bundle.entry.resource.ofType(Patient).extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding'
     condition: '%resource.code.exists()'
     value: [ '%resource.system.getCodingSystemMapping()' ]
-    hl7Spec: [ '%{PID}-10-3' ]
\ No newline at end of file
+    hl7Spec: [ '%{PID}-10-3' ]
+
+  - name: la-patient-county-codes
+    value: [ 'Bundle.entry.resource.ofType(Patient).address.district' ]
+    hl7Spec: [ '/PATIENT_RESULT/PATIENT/PID-11-6' ]
```
Comment (GilmoreA6) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml:67]: This should be PID-11-9. Make sure to validate the correct hl7 spec location for the value even if copying from the repo as the initial value could be incorrect (in this case me, I put in the wrong thing for AK)

Comment (JessicaWNava) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml:67]: Oh shoot, their email even specified that. Good catch!

**Review by GilmoreA6 — COMMENTED** [review]
Recommend adding an element that removes the county entirely if patient state not "LA" otherwise if you get a report with ordering facility state of "LA" and patient state of someone else the county won't be translated to FIPs code

**Comment by JessicaWNava** [comment]
> Recommend adding an element that removes the county entirely if patient state not "LA" otherwise if you get a report with ordering facility state of "LA" and patient state of someone else the county won't be translated to FIPs code

Decided to just do this right and create the fips lookup

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml [hunk 2]
```diff
@@ -31,13 +31,38 @@ elements:
...
+
   - name: la-patient-race-coding-system
     resource: 'Bundle.entry.resource.ofType(Patient).extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding'
     condition: '%resource.code.exists()'
     value: [ '%resource.system.getCodingSystemMapping()' ]
-    hl7Spec: [ '%{PID}-10-3' ]
\ No newline at end of file
+    hl7Spec: [ '%{PID}-10-3' ]
+
+  - name: la-patient-county-codes
```
Comment (GilmoreA6) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml:65]: This casues an exception if the lookup value can't be found:

"message":"gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchemaElementProcessingException: Error encountered while applying: la-patient-county-codes in /metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml to FHIR bundle. \nError was: convertToString(...) must not be null","thread":"pool-2-thread-38","timestamp":"2024-08-12T16:38:11.477Z","level":"ERROR","logger":"gov.cdc.prime.router.fhirengine.engine.FHIRTranslator"}
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] Executed 'Functions.translate-fhir' (Failed, Id=8dccbdf7-fa01-4d99-a372-20e91d92fb3e, Duration=633ms)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] System.Private.CoreLib: Exception while executing function: Functions.translate-fhir. System.Private.CoreLib: Result: Failure
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] Exception: NullPointerException: convertToString(...) must not be null
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] Stack: java.lang.reflect.InvocationTargetException
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.broker.JavaMethodInvokeInfo.invoke(JavaMethodInvokeInfo.java:22)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.broker.JavaMethodExecutor.execute(JavaMethodExecutor.java:22)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.chain.FunctionExecutionMiddleware.invoke(FunctionExecutionMiddleware.java:19)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.chain.InvocationChain.doNext(InvocationChain.java:21)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.invoke(JavaFunctionBroker.java:133)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.invokeMethod(JavaFunctionBroker.java:124)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:34)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:10)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.handler.MessageHandler.handle(MessageHandler.java:44)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at com.microsoft.azure.functions.worker.JavaWorkerClient$StreamingMessagePeer.lambda$onNext$0(JavaWorkerClient.java:94)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.lang.Thread.run(Thread.java:840)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] Caused by: gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchemaElementProcessingException: Error encountered while applying: la-patient-county-codes in /metadata/hl7_mapping/receivers/STLTs/LA/LA-receiver-transform.yml to FHIR bundle. 
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] Error was: convertToString(...) must not be null
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processSchema(FhirToHl7Converter.kt:164)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processSchema$default(FhirToHl7Converter.kt:142)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.process(FhirToHl7Converter.kt:90)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.engine.FHIRTranslator.getHL7MessageFromBundle$prime_router(FHIRTranslator.kt:227)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.engine.FHIRTranslator.getByteArrayFromBundle$prime_router(FHIRTranslator.kt:200)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.engine.FHIRTranslator.sendTranslated$prime_router(FHIRTranslator.kt:148)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.engine.FHIRTranslator.doWork(FHIRTranslator.kt:84)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.engine.FHIREngine.run(FHIREngine.kt:224)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions$runFhirEngine$newMessages$1.invoke(FHIRFunctions.kt:197)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions$runFhirEngine$newMessages$1.invoke(FHIRFunctions.kt:196)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning$lambda$1(DatabaseAccess.kt:107)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at org.jooq.impl.DefaultDSLContext.lambda$transactionResult0$3(DefaultDSLContext.java:531)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at org.jooq.impl.Tools$3$1.block(Tools.java:6068)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3465)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3436)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at org.jooq.impl.Tools$3.get(Tools.java:6065)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at org.jooq.impl.DefaultDSLContext.transactionResult0(DefaultDSLContext.java:579)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at org.jooq.impl.DefaultDSLContext.transactionResult(DefaultDSLContext.java:503)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning(DatabaseAccess.kt:107)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.runFhirEngine(FHIRFunctions.kt:196)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.doTranslate(FHIRFunctions.kt:169)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.doTranslate$default(FHIRFunctions.kt:163)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.translate(FHIRFunctions.kt:154)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  ... 19 more
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] Caused by: java.lang.NullPointerException: convertToString(...) must not be null
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirPathUtils.evaluateString(FhirPathUtils.kt:228)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.getValueAsString$prime_router(FhirToHl7Converter.kt:117)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processElement$prime_router(FhirToHl7Converter.kt:225)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processSchema(FhirToHl7Converter.kt:162)
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z]  ... 41 more
2024-08-12 09:38:11 [2024-08-12T16:38:11.479Z] .
2024-08-12 09:38:11 [2024-08-12T16:38:11.928Z] Executing 'Functions.translate-fhir' (Reason='New queue message detected on 'elr-fhir-translate'.', Id=3fab637b-36ff-4a49-b7bc-e3839f110e35)
2024-08-12 09:38:11 [2024-08-12T16:38:11.928Z] Trigger Details: MessageId: 96211148-bac7-4f64-af88-778ef36ab3eb, DequeueCount: 5, InsertedOn: 2024-08-12T16:38:07.000+00:00

File: prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt [hunk 3]
```diff
@@ -150,4 +155,30 @@ class CustomFhirPathFunctionTest {
             (result[0] as StringType).value
         ).isEqualTo(loincCode)
     }
+
```
Comment (GilmoreA6) [inline prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt:158]: Need to include tests for sadpath as well as happypath (i.e. what happens when value not found)

**Review by GilmoreA6 — APPROVED** [review]

**Pull request merged**